### PR TITLE
update cluster module to redux

### PR DESCRIPTION
**Pull request by garrensmith**
## Overview

Update Cluster module to use redux

## Testing recommendations

* Tests should pass. 
* n = 1 Able to create admin and change password
* n > 1 Show disable cluster page for config and admin page

## Checklist

- [x] Code is written and works correctly;
- [x] Changes are covered by tests;

### CODE REVIEW

File: app/addons/cluster/api.js [hunk 1]
```diff
@@ -10,29 +10,20 @@
...
-  }
-});
-
-export default Cluster;
+import {get} from '../../core/ajax';
+
+export default () => {
+  return get(Helpers.getServerUrl('/_membership'))
+    .then(res => {
+      const list = res.all_nodes.reduce(function (acc, node) {
```
Comment (Antonio-Maranhao) [inline app/addons/cluster/api.js]: Can you please add some error handling here:
```
      if (!res.all_nodes) {
        const details = res.reason ? res.reason : '';
        throw new Error('Failed to load list of nodes.' + details);
      }
```

File: app/addons/cluster/actions.js [hunk 2]
```diff
@@ -11,35 +11,31 @@
...
-    var memberships = new ClusterResources.ClusterNodes();
-
-    memberships.fetch().then(() => {
+  fetchNodes () {
+    getNodes().then((nodes) => {
       this.updateNodes({
-        nodes: memberships.get('nodes_mapped')
+        nodes: nodes.nodes_mapped
       });
     });
```
Comment (Antonio-Maranhao) [inline app/addons/cluster/actions.js]: Catch errors here:
```
}).catch(err => {
      FauxtonAPI.addNotification({
        type: 'error',
        msg: err.message,
        clear: true
      });
    });
```

File: app/addons/cluster/actions.js [hunk 3]
```diff
@@ -11,35 +11,31 @@
...
+    getNodes().then((nodes) => {
+      const allNodes = nodes.all_nodes;
 
-      if (nodes.length === 1) {
-        return FauxtonAPI.navigate(successtarget + nodes[0]);
+      if (allNodes.length === 1) {
+        return FauxtonAPI.navigate(successtarget + allNodes[0]);
       }
       return FauxtonAPI.navigate('/cluster/disabled', {trigger: true});
     });
```
Comment (Antonio-Maranhao) [inline app/addons/cluster/actions.js]: Same here: 
```
}).catch(err => {
      FauxtonAPI.addNotification({
        type: 'error',
        msg: err.message,
        clear: true
      });
    });
```

**Comment by garrensmith** [comment]
@Antonio-Maranhao good idea. I've added the error handling.

**Pull request merged**